crates/pathfinder/src/sequencer.rs

```diff
@@ -271,58 +137,22 @@ impl Client {
         })
     }
 
-    /// Helper function that constructs a URL for particular query.
-    fn build_query(&self, path_segments: &[&str], params: &[(&str, &str)]) -> Url {
-        let mut query_url = self.sequencer_url.clone();
-        query_url
-            .path_segments_mut()
-            .expect("Base URL is valid")
-            .extend(path_segments);
-        query_url.query_pairs_mut().extend_pairs(params);
-        tracing::trace!(%query_url);
-        query_url
+    fn request(&self) -> builder::Request<builder::stage::Gateway> {
```

Comment:
Shouldn't there be a lifetime on the return value?

Reply:
Appears to be ellided? Or rather, it hasn't forced me to add it 🤔

Reply:
Need to add the warning then on these. I guess too many write them out by default.

Reply:
[Tracking issue for warning for rust_2018_idioms by default ](https://github.com/rust-lang/rust/issues/54910) indicates they were being conservative. 

Personally, I didn't even know this was a thing :).

Reply:
I'll create a new PR which fixes the whole codebase and adds the check to ci.